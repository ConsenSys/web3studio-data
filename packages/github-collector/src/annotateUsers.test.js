const { from } = require('rxjs');
const { toArray, take, map } = require('rxjs/operators');
const collectRepos = require('./collectRepos');
const collectForks = require('./collectForks');
const annotateUsers = require('./annotateUsers');
const mockGithub = require('../test/fixtures/mockGithub');
const TestTransport = require('../test/TestTransport');

describe('annotateUsers', () => {
  let source;
  let transport;

  afterAll(mockGithub.afterAll);
  beforeAll(mockGithub.beforeAll);

  beforeEach(() => {
    transport = new TestTransport();

    mockGithub.beforeAll();

    source = collectRepos().pipe(take(1));
  });

  it("Adds data to users to show if they're in consensys, web3studio, or other", async () => {
    const results = await source
      .pipe(
        collectForks(transport),
        annotateUsers(),
        map(event => event.meta.user.group),
        toArray()
      )
      .toPromise();

    expect(results.length).toEqual(7);

    expect(results).toMatchSnapshot();
  });

  it("passes events that don't contain user data untouched", async () => {
    const event = {
      event: 'ci-passed',
      timetamp: Date.now(),
      meta: {
        commit: 'f10dcce'
      }
    };
    const results = await from([event])
      .pipe(annotateUsers())
      .toPromise();

    expect(results).toEqual(event);
  });
});
