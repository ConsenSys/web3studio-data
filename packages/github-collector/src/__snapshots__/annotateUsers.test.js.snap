// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`annotateUsers Adds data to users to show if they're in consensys, web3studio, or other 1`] = `
Array [
  "web3studio",
  "other",
  "web3studio",
  "web3studio",
  "consensys",
  "other",
  "other",
]
`;
